core/lib/src/rocket.rs

```diff
@@ -610,22 +750,26 @@ impl Rocket {
     ///     format!("The stateful value is: {}", state.0)
     /// }
     ///
-    /// fn main() {
+    /// #[rocket::main]
+    /// async fn main() {
     /// # if false { // We don't actually want to launch the server in an example.
     ///     rocket::ignite()
     ///         .mount("/", routes![index])
     ///         .manage(MyValue(10))
-    ///         .launch();
+    ///         .launch()
+    ///         .await;
     /// # }
     /// }
     /// ```
     #[inline]
-    pub fn manage<T: Send + Sync + 'static>(self, state: T) -> Self {
-        if !self.state.set::<T>(state) {
-            error!("State for this type is already being managed!");
-            panic!("Aborting due to duplicately managed state.");
-        }
-
+    pub fn manage<T: Send + Sync + 'static>(mut self, state: T) -> Self {
+        self.pending.push(BuildOperation::Manage(Box::new(|rocket| {
```

Comment:
Clever! Do we have tests that ensure this `panic` actually happens?

Reply:
I don't see such a test, but it's easy to add.

Reply:
While working on this I noticed some docs discrepancies: `mount` can panic immediately or on `finish`, `manage` won't panic until `finish`, and an `attach`ed `on_attach` fairing is no longer executed immediately. I'll make sure to address those.

Reply:
What's the status of the discrepancies that you found?